src/state/mod.rs

```diff
@@ -133,10 +133,10 @@ impl StateDiff {
         let state_cache = cached_state.cache().to_owned();
 
         let substracted_maps = state_cache.storage_writes;
-        let storage_updates = to_state_diff_storage_mapping(substracted_maps);
+        let storage_updates = to_state_diff_storage_mapping(&substracted_maps);
 
-        let address_to_nonce = state_cache.nonce_writes;
-        let class_hash_to_compiled_class = state_cache.compiled_class_hash_writes;
+        let address_to_nonce = state_cache.nonce_writes.clone();
```

Comment:
these 2 clones can have a huge impact in performance

Reply:
Removed